config.m4

```diff
@@ -111,6 +111,12 @@ if test "$MONGODB" != "no"; then
     PHP_CHECK_GCC_ARG(-Wdeclaration-after-statement,    _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wdeclaration-after-statement")
     dnl PHP_CHECK_GCC_ARG(-Werror,                          _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Werror")
 
+    PHP_MONGODB_FOUND_VERSION=`${PHP_CONFIG} --version`
+    PHP_MONGODB_FOUND_VERNUM=`echo "${PHP_MONGODB_FOUND_VERSION}" | $AWK 'BEGIN { FS = "."; } { printf "%d", ([$]1 * 100 + [$]2) * 100 + [$]3;}'`
+
+    if test "$PHP_MONGODB_FOUND_VERNUM" -ge "50400"; then
+      PHP_CHECK_GCC_ARG(-Wno-cast-qual,                 _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wno-cast-qual")
+    fi
```

Comment:
For reference, here are a few of these errors from 5.3:

```
/home/jmikola/workspace/mongodb/phpc/src/bson.c: In function ‘zif_fromPHP’:
/home/jmikola/workspace/mongodb/phpc/src/bson.c:967:55: warning: passing argument 3 of ‘zend_parse_parameters’ discards ‘const’ qualifier from pointer target type
  if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "A", &data) == FAILURE) {
                                                       ^
In file included from /home/jmikola/bin/php-bin/php-5.3/include/php/main/php.h:38:0,
                 from /home/jmikola/workspace/mongodb/phpc/src/bson.c:29:
/home/jmikola/bin/php-bin/php-5.3/include/php/Zend/zend_API.h:245:14: note: expected ‘char *’ but argument is of type ‘const char *’
 ZEND_API int zend_parse_parameters(int num_args TSRMLS_DC, char *type_spec, ...);
              ^
/home/jmikola/workspace/mongodb/phpc/src/bson.c:974:1: warning: passing argument 3 of ‘_estrndup’ discards ‘const’ qualifier from pointer target type
  RETVAL_STRINGL((const char *) bson_get_data(bson), bson->len, 1);
 ^
```

A full log of the stderr output from a 5.3 build is here: https://gist.github.com/jmikola/e3a3f7da387333187864